feat: enable BrowserWindow level placement on Windows

CapOM opened the pull request:
#### Description of Change

For now it only adds the ability to place the window below
the task bar while still being always on top.
Previous behaviour was always showing the window above the task
bar when top is true. We keep this default behaviour, i.e. when
the 'level' parameter is omitted.

https://github.com/electron/electron/issues/18933

#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Can set a window always on top but behind the taskbar on Windows

CapOM commented:
CC @codebytere

codebytere commented on docs/api/browser-window.md:
```diff
@@ -1102,8 +1102,14 @@ On Linux always returns `true`.
+  `screen-saver`, and ~~`dock`~~ (Deprecated). The default is `floating` when
+  `flag` is true. The `level` is reset to `normal` when the flag is false. See
+  the [macOS docs][window-levels] for more details.
+* `level` String (optional) _Windows_ - Values are the same as on macOS to make
```
This shouldn't be the a different parameter with the same name; that'll confuse our documentation parser. This should just specify what values of the existing `level` parameter now apply to Windows.

CapOM replied:
Done.

codebytere commented on shell/browser/native_window_views.cc:
```diff
@@ -757,6 +759,38 @@ void NativeWindowViews::SetAlwaysOnTop(bool top,
+#if defined(OS_WIN)
+  // Re-use level names from MacOS to have the same electron api.
+  if (top) {
+    bool above_task_bar = true;
```
We can probably make this logic cleaner by considering which `level` values change the default; something like: 
```cpp
if (top) {
    bool above_task_bar = true;
    std::vector<std::string> levels = {"floating", "torn-off-menu", "modal-panel", "main-menu", "status"};
    if (std::find(levels.begin(), levels.end(), value) != levels.end()) {
      above_task_bar = false;
    }
    
    if (!above_task_bar) {
      // Behind the taskbar.
      const HWND task_bar_hwnd = ::FindWindow(kUniqueTaskBarClassName, NULL);
      ::SetWindowPos(GetAcceleratedWidget(), task_bar_hwnd, 0, 0, 0, 0,
                    SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);
    }
}
```

We don't need to specifically check for the final three values ("pop-up-menu", "screen-saver", "dock") since they wouldn't change the default value you've already set at the top of the conditional.

CapOM replied:
Done

CapOM replied:
Thx a lot for the suggestion.

CapOM commented:
The test failures seem unrelated to this change. I addressed the remarks, please take a look, thx!

CapOM commented:
Hi, thx for the testing and suggestions. One could argue that the user of `setAlwaysOnTop` can still watch for the `focus` event and call this api again. But I am ok to internally ensure the `behind taskbar` property upon `NativeWindowViews::OnWidgetActivationChanged(when active is true)`, i.e. `focus`. I will also do it for `restore`, `resize` and `show`. That should cover most of the cases. Remaining corner cases will have to be handle for the user of this api. How does this plan sound ?

CapOM commented:
Hi @brenca , I addressed your remarks. Actually `OnWidgetActivationChanged` is internally called upon activate, restore and show so no need to ensure the placement else where.
Please take a look thx!

miniak commented on shell/browser/native_window_views.cc:
```diff
@@ -757,6 +759,20 @@ void NativeWindowViews::SetAlwaysOnTop(bool top,
+    const std::vector<std::string> levels = {
+        "floating", "torn-off-menu", "modal-panel", "main-menu", "status"};
+    behind_task_bar_ =
+        std::find(levels.begin(), levels.end(), level) != levels.end();
```
you can use `base::Contains` from `base/stl_util.h`

CapOM replied:
Done, thx

miniak commented on shell/browser/native_window_views.cc:
```diff
@@ -757,6 +759,20 @@ void NativeWindowViews::SetAlwaysOnTop(bool top,
+  if (top) {
+    // On macOS the window is placed behind the Dock for the following levels.
+    // Re-use the same names on Windows to make it easier for the user.
+    const std::vector<std::string> levels = {
```
don't allocate this on the heap unnecessarily
```
static const auto levels[] = {
        "floating", "torn-off-menu", "modal-panel", "main-menu", "status"};
```

CapOM replied:
Done

miniak commented on shell/browser/native_window_views.cc:
```diff
@@ -1346,6 +1364,17 @@ ui::WindowShowState NativeWindowViews::GetRestoredState() {
+void NativeWindowViews::MoveBehindTaskBarIfNeeded() {
+#if defined(OS_WIN)
+  if (behind_task_bar_) {
+    const HWND task_bar_hwnd = ::FindWindow(kUniqueTaskBarClassName, NULL);
```
use `nullptr`

CapOM replied:
Done

CapOM commented:
Hi @miniak , thx for the review, I addressed the remarks, please take a look, thx!

CapOM commented:
Hi @miniak , hi @brenca , gentle ping ?

codebytere commented:
```
autoUpdater behavior "before each" hook for "should have a valid code signing identity" - "before each" hook for "should have a valid code signing identity"
```
is an expected failure for fork PRs and does not block merge.
